src/main/java/me/profelements/dynatech/items/tools/AngelGem.java

```diff
@@ -46,11 +49,13 @@ public AngelGem(Category category, SlimefunItemStack item, RecipeType recipeType
     private ItemDropHandler onItemDrop() {
         return (e, p, itemEntity) -> {
             ItemStack item = itemEntity.getItemStack();
-            if (item != null && item.getType() == this.getItem().getType() && item.hasItemMeta() && isItem(item) && e.getPlayer().getGameMode() != GameMode.CREATIVE) {
+            if (SlimefunUtils.isItemSimilar(this.getItem(), DynaTechItems.ANGEL_GEM, false, false) && e.getPlayer().getGameMode() != GameMode.CREATIVE && item.getType() == DynaTechItems.ANGEL_GEM.getType()) {

```

Comment:
Less intensive code should be first, you're literally putting the most intensive call first

```suggestion
            if (e.getPlayer().getGameMode() != GameMode.CREATIVE && item.getType() == DynaTechItems.ANGEL_GEM.getType() && SlimefunUtils.isItemSimilar(this.getItem(), DynaTechItems.ANGEL_GEM, false, false)) {

```

Reply:
also, getByItem would be better here.